service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/C2UploadedEmailContentProvider.java

```diff
@@ -25,16 +24,16 @@
     private final ObjectMapper mapper;
     private final Time time;
 
-    public Map<String, Object> buildC2UploadNotification(final CaseDetails caseDetails) {
+    public C2UploadedTemplate buildC2UploadNotification(final CaseDetails caseDetails) {
         CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
-        final String subjectLine = buildSubjectLine(caseData.getFamilyManCaseNumber(), caseData.getRespondents1());
 
-        return ImmutableMap.<String, Object>builder()
-            .putAll(buildCommonNotificationParameters(caseDetails))
-            .put("subjectLine", subjectLine)
-            .put("hearingDetailsCallout", subjectLine)
-            .put("reference", String.valueOf(caseDetails.getId()))
-            .build();
+        C2UploadedTemplate adminTemplateForC2 = new C2UploadedTemplate();
+        adminTemplateForC2.setCallout(buildCallout(caseData));
+        adminTemplateForC2.setRespondentLastName(getFirstRespondentLastName(caseData.getRespondents1()));
+        adminTemplateForC2.setCaseUrl(getCaseUrl(caseDetails.getId()));
+        adminTemplateForC2.setDocumentLink(linkToAttachedDocument(caseData.getSubmittedForm()));
+
```

Comment:
We could use builder pattern here (my preference) but this is ok too.

Reply:
I think setter is fine so leaving as is